Wikipedia/Code/ArticleViewController.swift

```diff
@@ -556,41 +565,65 @@ class ArticleViewController: ThemeableViewController, HintPresenting, UIScrollVi
+            self.needsWatchButton = false
+            self.needsUnwatchFullButton = false
+            self.needsUnwatchHalfButton = false
+            self.toolbarController.updateMoreButton(needsWatchButton: self.needsWatchButton, needsUnwatchHalfButton: self.needsUnwatchHalfButton, needsUnwatchFullButton: self.needsUnwatchFullButton, previousArticleTab: self.previousArticleTab, nextArticleTab: self.nextArticleTab)
             return
         }
-                
-        WMFArticleDataController().fetchArticleInfo(title: title, project: project, request: request) { result in
+        
+        WMFArticleDataController().fetchArticleInfo(title: title, project: project, request: request) { [weak self] result in
             
-            DispatchQueue.main.async { [weak self] in
-                
-                guard let self else {
-                    return
-                }
+            guard let self else { return }
+            
+            switch result {
+            case .success(let info):
                 
-                switch result {
-                case .success(let info):
-                    
-                    let needsWatchButton = !info.watched
-                    let needsUnwatchHalfButton = info.watched && info.watchlistExpiry != nil
-                    let needsUnwatchFullButton = info.watched && info.watchlistExpiry == nil
+                DispatchQueue.main.async {
+                    self.needsWatchButton = !info.watched
+                    self.needsUnwatchHalfButton = info.watched && info.watchlistExpiry != nil
+                    self.needsUnwatchFullButton = info.watched && info.watchlistExpiry == nil
                     
-                    self.toolbarController.updateMoreButton(needsWatchButton: needsWatchButton, needsUnwatchHalfButton: needsUnwatchHalfButton, needsUnwatchFullButton: needsUnwatchFullButton)
+                    self.toolbarController.updateMoreButton(needsWatchButton: self.needsWatchButton, needsUnwatchHalfButton: self.needsUnwatchHalfButton, needsUnwatchFullButton: self.needsUnwatchFullButton, previousArticleTab: self.previousArticleTab, nextArticleTab: self.nextArticleTab)
                     
                     if needsCategories {
                         self.saveCategories(categories: info.categories, articleTitle: title, project: project)
                     }
-                    
-                case .failure(let error):
-                    DDLogError("Error fetching article MediaWiki info: \(error)")
                 }
+                
+            case .failure(let error):
+                DDLogError("Error fetching article MediaWiki info: \(error)")
+            }
+        }
+    }
+    
+    func loadNextAndPreviousArticleTabs() {
+        guard let title = articleURL.wmf_title,
+            let siteURL = articleURL.wmf_site,
+            let project = WikimediaProject(siteURL: siteURL)?.wmfProject else {
+                return
```

Comment:
`title` and `project` are not being used, you could use _ here so the warning goes away